react/src/components/home/Home.jsx

```diff
@@ -25,55 +25,185 @@ function Home() {
+  const [current, setCurrent] = useState({
+    lat: 34.0549,
+    lng: -118.2426,
+    active: "off",
+  });
   const [zoom, setZoom] = useState(15);
-  
+
+  const [resultsData, setResultsData] = useState({
+    arrayOfResults: results,
+    firstFilteredResults: [],
+    secondFilteredResults: [],
+  });
+
+  const [query, setQuery] = useState("");
+
+  const [locationFilter, setLocationFilter] = useState("");
+
+  const handleSearch = (e) => {
+    setQuery(e.target.value);
+  };
+
+  const handleLocation = (e) => {
+    setLocationFilter(e.target.value);
+  };
+
+  const onSearchRequested = (e) => {
+    e.preventDefault();
+
+    resultsData.firstFilteredResults = filterByValue(results, query);
+
+    if (locationFilter) {
+      resultsData.secondFilteredResults = filterByLocation(
+        resultsData.firstFilteredResults,
+        locationFilter
+      );
+    } else {
+      resultsData.secondFilteredResults = resultsData.firstFilteredResults;
+    }
+
+    setResultsData((prevState) => {
+      let newResults = resultsData.secondFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  const onLocationRequested = (e) => {
+    e.preventDefault();
+
+    resultsData.firstFilteredResults = filterByLocation(
+      results,
+      locationFilter
+    );
+
+    if (query) {
+      resultsData.secondFilteredResults = filterByValue(
+        resultsData.firstFilteredResults,
+        query
+      );
+    } else {
+      resultsData.secondFilteredResults = resultsData.firstFilteredResults;
+    }
+
+    setResultsData((prevState) => {
+      let newResults = resultsData.secondFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  const onResetQueryFilter = (e) => {
+    e.preventDefault();
+
+    setQuery("");
+    clearQueryField();
+
+    if (locationFilter) {
+      resultsData.firstFilteredResults = filterByLocation(
+        results,
+        locationFilter
+      );
+    } else if (!locationFilter) {
+      resultsData.firstFilteredResults = results;
+    }
+    setResultsData((prevState) => {
+      let newResults = resultsData.firstFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
+
+      return rd;
+    });
+  };
+
+  const onResetLocationFilter = (e) => {
+    e.preventDefault();
+
+    setLocationFilter("");
+    clearLocationField();
+
+    if (query) {
+      resultsData.firstFilteredResults = filterByValue(results, query);
+    } else if (!query) {
+      resultsData.firstFilteredResults = results;
+    }
+    setResultsData((prevState) => {
+      let newResults = resultsData.firstFilteredResults;
+
+      const rd = { ...prevState };
+      rd.arrayOfResults = newResults;
+
+      console.log(newResults);
```

Comment:
I think it might be a good idea to remove the console.logs for the PR.

Reply:
I also agree.